shared-libs/transitions/src/transitions/index.js

```diff
@@ -285,10 +168,15 @@ const finalize = ({ change, results }, callback) => {
  * change/write.
  */
 const applyTransition = ({ key, change, transition }, callback) => {
+  if (!canRun({ key, change, transition })) {
+    logger.debug(
+      `canRun test failed on transition ${transition.key} for doc ${change.id} seq ${change.seq}`
+    );
+    return callback();
+  }
+
   logger.debug(
```

Comment:
I moved the call to `canRun` from the `applyTransitions` map function.
The reason was to test `canRun` over the document after it has been updated by previous transitions.

As an example, it enables the possibility of running `registration` over a fresh doc: currently `registration` requires either the form to be public or `doc.contact` to exist, while `update_clinics` is responsible for creating `doc.contact`.